Src/Newtonsoft.Json.UnityConverters/Matrix4x4Converter.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using Newtonsoft.Json.Linq;
+using UnityEngine;
+
+namespace Newtonsoft.Json.UnityConverters
+{
+    public class Matrix4x4Converter : JsonConverter
+    {
+        public override void WriteJson(JsonWriter writer, object? value, JsonSerializer serializer)
+        {
+            if (value == null)
+            {
+                writer.WriteNull();
+                return;
+            }
+
+            var m = (Matrix4x4)value;
+
+            writer.WriteStartObject();
+
+            writer.WritePropertyName("m00");
+            writer.WriteValue(m.m00);
+
+            writer.WritePropertyName("m01");
+            writer.WriteValue(m.m01);
+
+            writer.WritePropertyName("m02");
+            writer.WriteValue(m.m02);
+
+            writer.WritePropertyName("m03");
+            writer.WriteValue(m.m03);
+
+            writer.WritePropertyName("m10");
+            writer.WriteValue(m.m10);
+
+            writer.WritePropertyName("m11");
+            writer.WriteValue(m.m11);
+
+            writer.WritePropertyName("m12");
+            writer.WriteValue(m.m12);
+
+            writer.WritePropertyName("m13");
+            writer.WriteValue(m.m13);
+
+            writer.WritePropertyName("m20");
+            writer.WriteValue(m.m20);
+
+            writer.WritePropertyName("m21");
+            writer.WriteValue(m.m21);
+
+            writer.WritePropertyName("m22");
+            writer.WriteValue(m.m22);
+
+            writer.WritePropertyName("m23");
+            writer.WriteValue(m.m23);
+
+            writer.WritePropertyName("m30");
+            writer.WriteValue(m.m30);
+
+            writer.WritePropertyName("m31");
+            writer.WriteValue(m.m31);
+
+            writer.WritePropertyName("m32");
+            writer.WriteValue(m.m32);
+
+            writer.WritePropertyName("m33");
+            writer.WriteValue(m.m33);
+
+            writer.WriteEnd();
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object? existingValue, JsonSerializer serializer)
+        {
+            if (reader.TokenType == JsonToken.Null)
+            {
+                return new Matrix4x4();
+            }
+
+            var obj = JObject.Load(reader);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'obj'.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity.Converters/pullRequest?prid=5190745)